### PR TITLE
Trip Monitor analyzer improvement, status log

### PR DESCRIPTION
**Pull request by binh-dam-ibigroup**
### Checklist

- [x] Appropriate branch selected _(all PRs must first be merged to `dev` before they can be merged to `master`)_
- [x] Any modified or new methods or classes have helpful JavaDoc and code is thoroughly commented
- [na] The description lists all applicable issues this PR seeks to resolve
- [na] The description lists any configuration setting(s) that differ from the default settings
- [x] All tests and CI builds passing

### Description

This PR improves the performance of the trip monitoring job (and reduces the risk of bugs from dropped Mongo connections) and adds logging to inform when trip analysis queues and states remain non-empty/non-idle for too long.

### CODE REVIEW

File: src/main/java/org/opentripplanner/middleware/tripmonitor/jobs/MonitorAllTripsJob.java [hunk 1]
```diff
@@ -46,22 +47,50 @@ public void run() {
...
-            for (MonitoredTrip monitoredTrip : Persistence.monitoredTrips.getAll()) {
-                // attempt to add trip to tripAnalysisQueue until a spot opens up in the queue. If the timeout is
+            // Request all monitored trip IDs from the Mongo collection, all at once, and loop through a list of strings.
+            // If we looped using a Mongo-provided iterator instead, and the Mongo connection is dropped for any reason
+            // while the iterator is open, this thread would become blocked and
+            // prevent subsequent trip monitoring jobs to start.
+            // Performance note: Don't retrieve the full data for each trip at this time.
+            // This saves bandwidth and memory, as we don't use the trip data immediately besides the ID.
+            // The full data for each trip will be fetched at the time the actual analysis takes place.
+            // TODO: Filter out trips that would be skipped by the CheckMonitoredTrip.
```
Comment (br648) [inline src/main/java/org/opentripplanner/middleware/tripmonitor/jobs/MonitorAllTripsJob.java]: Can this TODO be done now? If not, perhaps explain why. TODOs seem to have a habit of not getting done.

File: src/main/java/org/opentripplanner/middleware/tripmonitor/jobs/MonitorAllTripsJob.java [hunk 2]
```diff
@@ -46,22 +47,50 @@ public void run() {
...
             }
 
             // wait for queue to deplete
+            int queueIterations = 0;
             while (tripAnalysisQueue.size() > 0) {
                 Thread.sleep(BLOCKING_QUEUE_DEPLETE_WAIT_TIME_MILLIS);
+                queueIterations++;
+                // Report queue status every minute (unless this job finishes before).
+                int runMillis = queueIterations * BLOCKING_QUEUE_DEPLETE_WAIT_TIME_MILLIS;
+                if ((runMillis % 60000) == 0) {
```
Comment (br648) [inline src/main/java/org/opentripplanner/middleware/tripmonitor/jobs/MonitorAllTripsJob.java]: Could create a constant for 60000 and use here and further down.

File: src/main/java/org/opentripplanner/middleware/tripmonitor/jobs/MonitoredTripLocks.java [hunk 3]
```diff
@@ -17,27 +17,27 @@ public class MonitoredTripLocks {
     /** the amount of time in milliseconds to wait to check if a lock has been released */
     private static final int LOCK_CHECK_WAIT_MILLIS = 500;
 
-    private static final ConcurrentHashMap<MonitoredTrip, Boolean> locks = new ConcurrentHashMap();
+    private static final ConcurrentHashMap<String, Boolean> locks = new ConcurrentHashMap();
```
Comment (br648) [inline src/main/java/org/opentripplanner/middleware/tripmonitor/jobs/MonitoredTripLocks.java]: SonarLint: Update to `new ConcurrentHashMap<>()`

**Review by br648 — APPROVED** [review]
Good updates. Some minor comments, but approving.

File: src/main/java/org/opentripplanner/middleware/tripmonitor/jobs/MonitorAllTripsJob.java (hunk 1 above)
Comment (binh-dam-ibigroup) [reply src/main/java/org/opentripplanner/middleware/tripmonitor/jobs/MonitorAllTripsJob.java]: Yes, thank you, I added one criterion in https://github.com/ibi-group/otp-middleware/pull/199/commits/ca89702fd4441423f3af8a721054834fe7bc7d64 and reworded the TODO.

File: src/main/java/org/opentripplanner/middleware/tripmonitor/jobs/MonitorAllTripsJob.java (hunk 2 above)
Comment (binh-dam-ibigroup) [reply src/main/java/org/opentripplanner/middleware/tripmonitor/jobs/MonitorAllTripsJob.java]: Refactored in https://github.com/ibi-group/otp-middleware/pull/199/commits/ca89702fd4441423f3af8a721054834fe7bc7d64.

File: src/main/java/org/opentripplanner/middleware/tripmonitor/jobs/MonitoredTripLocks.java (hunk 3 above)
Comment (binh-dam-ibigroup) [reply src/main/java/org/opentripplanner/middleware/tripmonitor/jobs/MonitoredTripLocks.java]: Refactored in https://github.com/ibi-group/otp-middleware/pull/199/commits/ca89702fd4441423f3af8a721054834fe7bc7d64.

**Review by JymDyerIBI — APPROVED** [review]
👍  Looks good.

**Pull request merged**